controllers/spec/enforcer/resources_count_spec/statefulset/ReplicaDbCountSpecEnforcer.go

```diff
@@ -22,35 +22,164 @@ package statefulset
+
 func CreateReplicaDbCountSpecEnforcer(
-	kubegresContext ctx.KubegresContext,
+	kubegresContext kubegresCtx.KubegresContext,
 	resourcesStates states.ResourcesStates,
 	resourcesCreator template.ResourcesCreatorFromTemplate,
-	blockingOperation *operation.BlockingOperation) ReplicaDbCountSpecEnforcer {
+	blockingOperation *operation.BlockingOperation,
+	clusterName string,
+) (ReplicaDbCountSpecEnforcer, error) {
+
+	enforcer := ReplicaDbCountSpecEnforcer{
+		kubegresContext:               kubegresContext,
+		resourcesStates:               resourcesStates,
+		resourcesCreator:              resourcesCreator,
+		blockingOperation:             blockingOperation,
+		replicationSlotsCreateDeleter: noopReplicationSlotsCreateDeleter{},
+	}
+
+	// TODO(piotrkpc): should this be here ? not testable code really
+	if kubegresContext.Kubegres.Spec.ReplicationSlots.Enabled {
+		sqlConn, ok := kubegresContext.GetSQLConnection()
+		if !ok {
+			return ReplicaDbCountSpecEnforcer{}, errors.New("get SQL connection from kubegresContext")
+		}
+		enforcer.replicationSlotsCreateDeleter = newSimpleReplicationSlotsCreateDeleter(
+			kubegresContext,
+			resourcesStates,
+			replicationSlotRepo.New(sqlConn.DB()),
+			clusterName,
+		)
+	}
+
+	return enforcer, nil
+}
+
+type simpleReplicationSlotCreateDeleter struct {
+	repo            replicationSlotRepo.Repository
+	states          states.ResourcesStates
+	kubegresContext kubegresCtx.KubegresContext
+	clusterName     string
+}
+
+func (r *simpleReplicationSlotCreateDeleter) Create(statefulSet *v1.StatefulSet) (*v1.StatefulSet, error) {
+	replicationSlotName, err := createReplicationSlotName(r.kubegresContext.Kubegres.GetName(), r.clusterName, statefulSet.GetName(), r.kubegresContext.ClusterRole())
+	objKey := ctrlclient.ObjectKeyFromObject(statefulSet)
+	if err != nil {
+		r.kubegresContext.Log.ErrorEvent("ReplicationSlotCreate", fmt.Errorf("replication slot name: %v: %w", objKey, err), "Failed to create replication slot name")
+		return nil, err
+	}
+	slot, err := r.repo.CreateSlot(r.kubegresContext.Ctx, replicationSlotName)
+	if err != nil {
+		r.kubegresContext.Log.ErrorEvent("ReplicationSlotCreate", fmt.Errorf("replication slot: %v for statefulSet: %v: %w", slot, objKey, err), "Failed to create replication slot")
+		return nil, err
+	}
+
+	ss := statefulSet.DeepCopy()
+	for i, container := range ss.Spec.Template.Spec.Containers {
+		ss.Spec.Template.Spec.Containers[i].Env = append(container.Env, corev1.EnvVar{
+			Name:  kubegresCtx.EnvVarReplicationSlotName,
+			Value: slot.Name,
+		})
+	}
+	for i, container := range ss.Spec.Template.Spec.InitContainers {
+		ss.Spec.Template.Spec.InitContainers[i].Env = append(container.Env, corev1.EnvVar{
+			Name:  kubegresCtx.EnvVarReplicationSlotName,
+			Value: slot.Name,
+		})
+	}
+	return ss, nil
+}
+
+func createReplicationSlotName(kubegresName, clusterName, statefulSetName string, role kubegresCtx.ClusterRole) (string, error) {
```

Comment:
nit: rename to `build` to avoid confusion with the replication slot creation.
```suggestion
func buildReplicationSlotName(kubegresName, clusterName, statefulSetName string, role kubegresCtx.ClusterRole) (string, error) {
```